tests/tt_eager/python_api_testing/unit_testing/misc/test_matmul_dram_sharded.py

```diff
@@ -0,0 +1,253 @@
+# SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+from loguru import logger
+import tt_lib as ttl
+from models.utility_functions import is_wormhole_b0, is_grayskull, skip_for_wormhole_b0
+from models.utility_functions import torch2tt_tensor, tt2torch_tensor, pad_by_zero, roundup32
+import torch
+from tests.tt_eager.python_api_testing.sweep_tests.comparison_funcs import (
+    comp_equal,
+    comp_pcc,
+)
+from tt_lib.utils import (
+    pad_weight,
+    tilize_to_list,
+    untilize,
+    is_close,
+)
+
+
+def find_max_subblock(out_block_h, out_block_w):
+    max_product = 0
+    best_h = 1
+    best_w = 1
+
+    for h in range(1, out_block_h + 1):
+        if out_block_h % h == 0:
+            for w in range(1, out_block_w + 1):
+                if out_block_w % w == 0 and h * w <= 8:
+                    if h * w > max_product:
+                        max_product = h * w
+                        best_h = h
+                        best_w = w
+    if out_block_w > best_w:
+        best_h = 1
+    return best_h, best_w, max_product
+
+
+def pad_to_dram_banks(num, lcm=32 * 12):
+    remainder = num % lcm
+    if remainder == 0:
+        return num
+    padding_needed = lcm - remainder
+    padded_number = num + padding_needed
+    return padded_number
+
+
+@pytest.mark.parametrize(
+    "fidelity",
+    [
+        ttl.tensor.MathFidelity.HiFi2,
+        ttl.tensor.MathFidelity.LoFi,
+    ],
+    ids=["HiFi2", "LoFi"],
+)
+@pytest.mark.parametrize(
+    "has_bias",
+    [
+        False,
+        True,
+    ],
+    ids=["no_bias", "bias"],
+)
+@pytest.mark.parametrize(
+    "in0_dtype, in1_dtype, out_dtype",
+    [
+        (ttl.tensor.DataType.BFLOAT16, ttl.tensor.DataType.BFLOAT8_B, ttl.tensor.DataType.BFLOAT16),
+    ],
+)
+@pytest.mark.parametrize(
+    "in1_in_dram, out_sharded, in0_sharded, M, K, N, activation, grid_size",
+    # "in1_in_dram, out_sharded, in0_sharded, M, K, N, activation, grid_size, in0_dtype, in1_dtype, out_dtype",
+    [
+        (False, True, True, 32, 8192, 1280, None, (8, 1)),
+        (False, True, True, 32, 8192, 4096, None, (8, 2)),
+        (False, True, True, 32, 8192, 1024, None, (8, 1)),
+        (False, True, True, 32, 32768, 1024, None, (8, 2)),
+        # (False, True, True, 32, 4096, 6144, None, (8, 2), ttl.tensor.DataType.BFLOAT16, ttl.tensor.DataType.BFLOAT8_B, ttl.tensor.DataType.BFLOAT16),
+        # (False, True, True, 32, 4096, 14336, None, (8, 2), ttl.tensor.DataType.BFLOAT16, ttl.tensor.DataType.BFLOAT4_B, ttl.tensor.DataType.BFLOAT8_B),
+        # (False, True, True, 32, 14336, 4096, None, (8, 2), ttl.tensor.DataType.BFLOAT8_B, ttl.tensor.DataType.BFLOAT8_B, ttl.tensor.DataType.BFLOAT8_B),
+        # (False, True, True, 32, 4096, 14336, None, (8, 2), ttl.tensor.DataType.BFLOAT16, ttl.tensor.DataType.BFLOAT4_B, ttl.tensor.DataType.BFLOAT8_B),
+    ],
+)
```

Comment:
Clean this up

Reply:
will leave it here for now, in case more benchmarking needed